input/core/src/commonMain/kotlin/majestic/calendar/tools/DateRangeState.kt

```diff
@@ -0,0 +1,71 @@
+package majestic.calendar.tools
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.Stable
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import kotlinx.datetime.LocalDate
+
+data class DateRangePickerLabels(
+    val from: String,
+    val to: String
+)
+
+@Stable
+class DateRangeState(
+    initialFromDate: LocalDate? = null,
+    initialToDate: LocalDate? = null,
+    val currentDate: LocalDate,
+    labels: DateRangePickerLabels
```

Comment:
when designing a public API, you have to be really aware of how the code is going to look at the call site

This code will look like this

```
DateRangeState(
  initialFromDate = LocalDate(2021,1,1),
  initialToDate = LocalDate(2021,2,1),
  currentDate: LocalDate.today()
)
```

instead of something as clean as

```
DateRangeState(from = LocalDate(2021,1,1), to = LocalDate(2021,2,1))
```